mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -352,61 +452,79 @@ def list_artifacts(self, path=None):
         return infos
 
     def _download_file(self, remote_file_path, local_path):
-        run_relative_remote_file_path = posixpath.join(
-            self.run_relative_artifact_repo_root_path, remote_file_path
-        )
-        read_credentials = self._get_read_credentials(self.run_id, run_relative_remote_file_path)
-        self._download_from_cloud(read_credentials.credentials, local_path)
+        """
+        _download_file is unused in this repository's implementation and is only defined out
+        of necessity because it is an abstract method in the `ArtifactRepository` base class
+        """
 
     def download_artifacts(self, artifact_path, dst_path=None):
         """
         Parallelized implementation of `download_artifacts` for Databricks.
         """
 
-        # Represents one or more in-progress artifact downloads to a local filesystem location
-        InflightDownloads = namedtuple(
-            "InflightDownloads",
+        # Represents an in-progress file artifact download to a local filesystem location
+        InflightDownload = namedtuple(
+            "InflightDownload",
             [
+                # The artifact path, given relative to the repository's artifact root location
+                "src_artifact_path",
                 # The local filesystem destination path to which artifacts are being downloaded
-                "local_dst_path",
-                # A map from artifact source paths, given relative to the repository's artifact
-                # root location, to futures representing their corresponding download operations
-                "src_paths_to_futures_map",
+                "dst_local_path",
+                # A future representing the artifact download operation
+                "download_future",
             ],
         )
 
-        def download_artifact(src_artifact_path, dst_local_dir_path):
+        def async_download_file_artifacts_from_paths(src_artifact_paths, dst_local_dir_path):
```

Comment:
file and directory downloads now rely on this method, which accepts a collection of paths to be downloaded and returns a collection of inflight download objects, each of which specifies the source artifact path, destination local filesystem path, and future.

Directory artifact downloads call this for each level in the artifact directory tree, and file artifact downloads call this with a single file input.